xcodebuild/xcodebuild.go

```diff
@@ -0,0 +1,73 @@
+package xcodebuild
+
+import (
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-xcode/models"
+	"github.com/bitrise-io/go-xcode/utility"
+	"github.com/bitrise-steplib/steps-xcode-test/fileremover"
+)
+
+// Xcodebuild ....
+type Xcodebuild interface {
+	RunBuild(buildParams Params, outputTool string) (string, int, error)
+	RunTest(params TestRunParams) (string, int, error)
+	Version() (Version, error)
+}
+
+type xcodebuild struct {
+	logger         log.Logger
+	commandFactory command.Factory
+	pathChecker    pathutil.PathChecker
+	fileRemover    fileremover.FileRemover
+}
+
+// New ...
+func New(logger log.Logger, commandFactory command.Factory, pathChecker pathutil.PathChecker, fileRemover fileremover.FileRemover) Xcodebuild {
+	return &xcodebuild{
+		logger:         logger,
+		commandFactory: commandFactory,
+		pathChecker:    pathChecker,
+		fileRemover:    fileRemover,
+	}
+}
+
+// Version ...
+type Version models.XcodebuildVersionModel
```

Comment:
Can we re-order the file so that `type` declarations are together at the top of the file?

Reply:
Isn't it better to keep it close to its usage?

Reply:
I'd vote for all types on the top. (At least I always look for them there). Not sure about golang conventions tho

Reply:
As discussed we will get back to this topic later.